nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyArrayTests.java

```diff
@@ -135,6 +135,40 @@ public void arrayContentsAndTopLevelAnnotation() {
         .doTest();
   }
 
+  @Test
+  public void nullableAssignmentNonnullArray() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static String [] foo = new String[10];",
+            "  static void foo() {",
+            "    // BUG: Diagnostic contains: assigning @Nullable expression to @NonNull field",
+            "    foo[1] = null;",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void nullableAssignmentNullableArray() {
```

Comment:
These tests look good!  For sanity, let's test a couple more scenarios:

1. The array should be a local variable
2. The array should be a method parameter

Reply:
Added!